### PR TITLE
Test/mento upgrade fork test

### LINKED ISSUE #7: Add Mento Upgrade integration tests
**Description**

We need to add integration tests that simulate the typical changes that would be made in a Mento upgrade proposal. The scenarios that should be tested for are as follows:

- [ ]  Deploy and init a new stable token

[Mento Upgrade Tests](https://www.notion.so/mentolabs/Governance-Test-Plan-4c38412bebe74095b025b26d2dc34ac0?pvs=4#13d5a1c3e97a48108e19d2a6444ddc4f)

**Acceptance Criteria**

- [ ] **Add integration tests for Mento Upgrades**
  - The integration tests should cover all the upgrade scenarios as defined in the test plan
  - The integration tests should be able to run successfully without any errors.

### PR DESCRIPTION
**Pull request by philbow61**
### Description

This PR adds tests for an example Mento Upgrade.
As we discussed in our grooming session we don't test an entire upgrade like MU03 but just deploying/initialising a new token + making changes to an existing mento Contract(Reserve).

### Other changes

For deploying the new StableToken I had to add the flattened contract to this repo.
I also made changes to the SDK to export not just the Mento governance addresses but also the other Mento contract addresses. therefore we are using 0.2.3 of the mento sdk

### Related issues

- Fixes #7

### CODE REVIEW

File: test/MentoUpgradeIntegration.test.ts [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      provider,
+    );
+    const reserveImplementation = await reserveProxy._getImplementation!();
+
+    const reserveImplementationContract = Reserve__factory.connect(
+      reserveImplementation,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      provider as any,
+    );
+    await reserveImplementationContract
```
[the remark this answers is not in the document]
Comment (philbow61) [reply test/MentoUpgradeIntegration.test.ts:224]: So I am transferring the ownership of both proxy and implementation as this is the state mento governance would be in on mainnet. On Alfjores I had the same problem with the implementation not being owned by celo governance, we were sloppy when we deployed mento V2 on testnets and didn't transfer ownership for both. 
Let me make sure celo governance owns both also on Baklava.

File: test/MentoUpgradeIntegration.test.ts [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    console.log(
+      'Running Mento Upgrade tests on network with chain id:',
+      chainId,
+    );
+    console.log('========================\r\n');
+  });
+
+  it('should allow for new Contracts to be initialized', async function () {
+    const newStableToken = await deployContract('StableTokenV2', [false]);
+    await newStableToken.transferOwnership!(mentoAddresses.TimelockController);
```
[the remark this answers is not in the document]
Comment (philbow61) [reply test/MentoUpgradeIntegration.test.ts:112]: I think I saw some errors in the ci, that's why I added them but might be unnecessary
https://github.com/mento-protocol/governance-tests/actions/runs/7873269222/job/21480228485

File: test/MentoUpgradeIntegration.test.ts [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+    timeTravel(7);
+
+    await mentoGovernor.connect(proposer)['queue(uint256)'](proposalId);
+
+    timeTravel(2);
+
+    await mentoGovernor.connect(proposer)['execute(uint256)'](proposalId);
+
+    expect(await newStableToken.name!()).to.equal('testToken');
```
[the remark this answers is not in the document]
Comment (philbow61) [reply test/MentoUpgradeIntegration.test.ts:151]: I think I saw some errors in the ci, that's why I added them but might be unnecessary
https://github.com/mento-protocol/governance-tests/actions/runs/7873269222/job/21480228485

File: test/MentoUpgradeIntegration.test.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    );
+  });
+
+  it('should allow for existing Contracts to be changed', async function () {
+    const target = mentoAddresses.Reserve;
+
+    const value = 0n;
+    // add a new stableToken to the reserve
+    const calldata1 = reserve.interface.encodeFunctionData('addToken', [
+      random.address,
```
[the remark this answers is not in the document]
Comment (philbow61) [reply test/MentoUpgradeIntegration.test.ts:164]: sure 👍

File: test/MentoUpgradeIntegration.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (philbow61) [reply test/MentoUpgradeIntegration.test.ts:223]: test now also passes on baklava

**Review by baroooo — APPROVED** [review]

**Pull request merged**